streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -269,68 +247,39 @@ public synchronized void put(final Bytes key,
         return originalValue;
     }
 
-    private void restoreAllInternal(final Collection<KeyValue<byte[], byte[]>> records) {
+    @Override
+    public void putAll(final List<KeyValue<Bytes, byte[]>> entries) {
         try (final WriteBatch batch = new WriteBatch()) {
-            for (final KeyValue<byte[], byte[]> record : records) {
-                if (record.value == null) {
-                    batch.delete(record.key);
-                } else {
-                    batch.put(record.key, record.value);
-                }
-            }
+            dbAccessor.prepareBatch(entries, batch);
             write(batch);
         } catch (final RocksDBException e) {
-            throw new ProcessorStateException("Error restoring batch to store " + name, e);
-        }
-    }
-
-    private void putInternal(final byte[] rawKey,
-                             final byte[] rawValue) {
-        if (rawValue == null) {
-            try {
-                db.delete(wOptions, rawKey);
-            } catch (final RocksDBException e) {
-                // String format is happening in wrapping stores. So formatted message is thrown from wrapping stores.
-                throw new ProcessorStateException("Error while removing key %s from store " + name, e);
-            }
-        } else {
-            try {
-                db.put(wOptions, rawKey, rawValue);
-            } catch (final RocksDBException e) {
-                // String format is happening in wrapping stores. So formatted message is thrown from wrapping stores.
-                throw new ProcessorStateException("Error while putting key %s value %s into store " + name, e);
-            }
+            throw new ProcessorStateException("Error while batch writing to store " + name, e);
         }
     }
 
-    void write(final WriteBatch batch) throws RocksDBException {
-        db.write(wOptions, batch);
-    }
-
     @Override
-    public void putAll(final List<KeyValue<Bytes, byte[]>> entries) {
-        try (final WriteBatch batch = new WriteBatch()) {
-            for (final KeyValue<Bytes, byte[]> entry : entries) {
-                Objects.requireNonNull(entry.key, "key cannot be null");
-                if (entry.value == null) {
-                    batch.delete(entry.key.get());
-                } else {
-                    batch.put(entry.key.get(), entry.value);
-                }
-            }
-            write(batch);
+    public synchronized byte[] get(final Bytes key) {
+        validateStoreOpen();
+        try {
+            return dbAccessor.get(key.get());
```

Comment:
Fair question. If we move all try-catch into the dbAccessor, we need to duplicate more code. For example, `db.flush()` could throw and catching the exception in `RocksDB#flush()` is a single place -- if we move it into the dbAccessor, we need to to try-catch in both implementations.

Thoughts?

Reply:
`put` is different, because you requested to catch `put()` and `get()` within `DualAccessor` separately -- thus, I moved the code inside.

For `putAll()` does also throw (main pattern is that the caller catches).